datafusion/physical-plan/src/sorts/cursor.rs

```diff
@@ -284,6 +333,34 @@ impl<T: FieldValues> Cursor for FieldCursor<T> {
         self.offset += 1;
         t
     }
+
+    fn slice(&self, offset: usize, length: usize) -> Self {
```

Comment:
What would happen if this method was simply

```
Self {
    values: self.values.slice(0, self.offset + offset + length),
    offset: self.offset + offset
    null_threshold: self.null_threshold,
}

```

Reply:
The RowCursor slicing does not slice the underlying rows, therefore it is `self.offset + offset`.
Whereas the FieldCursor does slice the underlying data, and therefore the offset is reset to 0.

Reply:
But the logic below simply ignores the value of `self.offset`?